From 7a735df586ec3a84d08bd334a8760bfea97101f3 Mon Sep 17 00:00:00 2001
From: Lingyun Cai <lingyun.cai@intel.com>
Date: Tue, 14 May 2024 08:52:01 +0000
Subject: [PATCH] Fix potential crash when accessing SpaceSplitString iterator

This CL fixes a potential crash when accessing the iterator with
data_ == nullptr.

Change-Id: I2e4bfb3926bb28d96b438c89841eed75be4d0a7b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5537048
Reviewed-by: Steinar H Gunderson <sesse@chromium.org>
Reviewed-by: Anders Hartvoll Ruud <andruud@chromium.org>
Commit-Queue: Steinar H Gunderson <sesse@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1300505}
---

--- a/third_party/blink/renderer/core/dom/space_split_string.h
+++ b/third_party/blink/renderer/core/dom/space_split_string.h
@@ -64,9 +64,11 @@
   bool IsNull() const { return !data_; }
   const AtomicString& operator[](wtf_size_t i) const { return (*data_)[i]; }
   Vector<AtomicString, 4>::const_iterator begin() const {
-    return data_->begin();
+    return data_ ? data_->begin() : nullptr;
+  }
+  Vector<AtomicString, 4>::const_iterator end() const {
+    return data_ ? data_->end() : nullptr;
   }
-  Vector<AtomicString, 4>::const_iterator end() const { return data_->end(); }
 
  private:
   class CORE_EXPORT Data : public RefCounted<Data> {
